packages/components/src/public/files/TemplateDownloadButton.tsx

```diff
@@ -1,35 +1,143 @@
+
+    const fetchTemplates = useCallback(async () => {
+        if (customTemplates || !isGridRenderer) return;
+        const hasCustomTemplates = await loadTemplates();
+        if (!hasCustomTemplates) {
+            if (onDownloadDefault)
+                onDownloadDefault();
+            else
+                downloadAttachment(defaultTemplateUrl, true);
+        }
+    }, [isGridRenderer, onDownloadDefault, customTemplates, setLoadingTemplates, defaultTemplateUrl]);
+
+    const dropdownTitle = useMemo(() => {
+        return (
+            <span title="Download Template">
+                <span className="fa fa-download" /> {text}
+            </span>
+        );
+    }, [text]);
+
+    if (!onDownloadDefault && !defaultTemplateUrl?.length && !showDropdown) return null;
 
     return (
         <RequiresPermission perms={[PermissionTypes.Insert, PermissionTypes.Update]} permissionCheck="any" user={user}>
-            <a
-                className={'btn btn-info ' + className}
-                title="Download Template"
-                onClick={onClick}
-                href={templateUrl}
-                rel="noopener noreferrer"
-                target="_blank"
-            >
-                <span className="fa fa-download" /> {text}
-            </a>
+            {!showDropdown && (
+                <a
+                    className={'btn btn-info ' + (className ?? '')}
+                    title="Download Template"
+                    onClick={onDownloadDefault}
+                    href={defaultTemplateUrl}
+                    rel="noopener noreferrer"
+                    target="_blank"
+                >
+                    <span className="fa fa-download" /> {text}
+                </a>
+            )}
+            {showDropdown && (
+                <DropdownButton
+                    onClick={fetchTemplates}
+                    title={dropdownTitle}
+                    bsStyle="info"
+                    noCaret={!customTemplates || customTemplates.length === 0}
+                    className="small-right-spacing"
+                    buttonClassName={dropDownClassName ?? (isGridRenderer ? 'button-small-padding' : '')}
+                >
+                    {customTemplates?.length > 0 && (
+                        <MenuItem
+                            key={0}
+                            href={defaultTemplateUrl}
+                            onClick={onDownloadDefault}
+                            rel="noopener noreferrer"
+                            target="_blank"
+                        >
+                            Default Template
+                        </MenuItem>
+                    )}
+                    {loadingTemplates && <LoadingSpinner />}
+                    {customTemplates?.map((template, ind) => {
+                        if (template.url.endsWith('(unavailable)'))
+                            return (
+                                <MenuItem disabled key={ind + 1} title="File not found">
```

Comment:
could we use `DisableableMenuItem` here so that the disabled message hover popover matches other menus?